modin/pandas/test/dataframe/test_iter.py

```diff
@@ -289,6 +290,28 @@ def test___setattr__mutating_column():
     df_equals(modin_df, pandas_df)
     assert modin_df.col0.equals(modin_df["col0"])
 
+    # Check that adding a new col via attributes raises warning
+    with pytest.warns(
+        UserWarning,
+        match="Modin doesn't allow columns to be created via a new attribute name - see",
+    ):
+        modin_df.col1 = [4]
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        modin_df.col1 = [5]
+        modin_df.col0 = 6
```

Comment:
What does this bit do? Looks like this test is failing here in CI

Reply:
This was originally suggested by @RehanSD. I think the idea is to make sure that these statements are not raising warnings. The test does pass locally using pytest. I'm not sure why it doesn't in CI. I removed one of the checks in the last commit, but it's still failing. I'm not yet sure why honestly.

Reply:
Oh I think I might know why. I think the base execution engine is Ray. The test you are failing uses a base execution engine of python so it's failing because it throws a warning that "we are defaulting to pandas". You can try running this locally yourself: `pytest -n 2 modin/pandas/test/dataframe/test_iter.py --execution=BaseOnPython`

Reply:
I think this is fixed now by matching on the warning message.